force-app/main/default/classes/JobApplicationTriggerHandler.cls

```diff
@@ -0,0 +1,502 @@
+                    Description = description, 
+                    WhatId = newJobApplication.Id, 
+                    WhoId = newJobApplication.Primary_Contact__c, 
+                    OwnerId = newJobApplication.OwnerId
+                );
+
+                tasksToInsert.add(taskToInsert);
+            }
+        }
+
+        insert tasksToInsert;
+    }
+
+    public void createTaskWhenStatusIsAccepted_Update(Map<Id, Job_Application__c> triggerDotOldMap, List<Job_Application__c> triggerDotNew) {
+        List<Task> tasksToInsert = new List<Task>();
+
+        for (Job_Application__c newJobApplication : triggerDotNew) {
+            if (
+                triggerDotOldMap.get(newJobApplication.Id).Status__c != newJobApplication.Status__c 
+                && newJobApplication.Status__c == 'Accepted' 
+                && triggerDotOldMap.get(newJobApplication.Id).Status__c != 'Closed'
+            ) {
+                String description = 
+                    '• Plan your resignation if applicable\n' 
+                    + '• Take some time to relax and recharge\n' 
+                    + '• Prepare for your first day of onboarding'
+                ;
+
+                Task taskToInsert = new Task(
+                    Subject = 'Celebration', 
+                    ActivityDate = Date.today().addDays(7), 
+                    Priority = 'Low', 
+                    Description = description, 
+                    WhatId = newJobApplication.Id, 
+                    WhoId = newJobApplication.Primary_Contact__c, 
+                    OwnerId = newJobApplication.OwnerId
+                );
+
+                tasksToInsert.add(taskToInsert);
+            }
+        }
+
+        insert tasksToInsert;
+    }
+
+    public void createTaskWhenStatusIsClosed_Insert(List<Job_Application__c> triggerDotNew) {
+        List<Task> tasksToInsert = new List<Task>();
+
+        for (Job_Application__c newJobApplication : triggerDotNew) {
+            if (newJobApplication.Status__c == 'Closed') {
+                String description = 
+                    '• Send a follow-up email thanking the interviewer and asking for feedback\n' 
+                    + '• Review your notes and reflect on areas of improvement'
+                ;
+
+                Task taskToInsert = new Task(
+                    Subject = 'Closed job application', 
+                    ActivityDate = Date.today().addDays(7), 
+                    Priority = 'Normal', 
+                    Description = description, 
+                    WhatId = newJobApplication.Id, 
+                    WhoId = newJobApplication.Primary_Contact__c, 
+                    OwnerId = newJobApplication.OwnerId
+                );
+
+                tasksToInsert.add(taskToInsert);
+            }
+        }
+
+        insert tasksToInsert;
+    }
+
+    public void createTaskWhenStatusIsClosed_Update(Map<Id, Job_Application__c> triggerDotOldMap, List<Job_Application__c> triggerDotNew) {
```

Comment:
How might you update this to reduce the amount of code to insert the various tasks?